pandas/io/parsers.py

```diff
@@ -40,70 +48,79 @@ class ParserWarning(Warning):
 
 Parameters
 ----------
-filepath_or_buffer : string or file handle / StringIO
-    The string could be a URL. Valid URL schemes include
-    http, ftp, s3, and file. For file URLs, a
-    host is expected. For instance, a local file could be
+filepath_or_buffer : str, pathlib.Path, py._path.local.LocalPath or any object \
```

Comment:
Note the '\' allows all the options to both obey the line character limit and be parsed into the html docs correctly. I'm not sure if it's a style we want or not.